datadog_checks_downloader/tests/test_downloader.py

```diff
@@ -59,26 +60,47 @@
 ]
 
 
-def _do_run_downloader(argv):
-    """Run the Datadog checks downloader."""
+@contextmanager
+def modified_args(argv):
     old_sys_argv = sys.argv
 
     sys.argv = ["datadog_checks_downloader"] + argv  # Make sure argv[0] (program name) is prepended.
-    try:
+
+    yield
+
+    sys.argv = old_sys_argv
+
+
+def _do_run_downloader(argv):
+    """Run the Datadog checks downloader."""
+
+    with modified_args(argv):
         download()
-    finally:
-        sys.argv = old_sys_argv
 
 
 @pytest.mark.online
-def test_download(capfd, distribution_name, distribution_version, temporary_local_repo, disable_verification):
+def test_download(capfd, distribution_name, distribution_version, temporary_local_repo, disable_verification, mocker):
     """Test datadog-checks-downloader successfully downloads and validates a wheel file."""
     argv = [distribution_name, "--version", distribution_version]
 
     if disable_verification:
         argv.append('--unsafe-disable-verification')
 
-    _do_run_downloader(argv)
+    with modified_args(argv):
+        tuf_downloader, standard_distribution_name, version, ignore_python_version = instantiate_downloader()
+
+        spy_with_tuf = mocker.spy(tuf_downloader, '_download_with_tuf')
```

Comment:
Almost there: what this should really test is that `_download_with_tuf_in_toto` was called or not called.

This means that `_download_without_tuf` should be renamed to `_download_without_tuf_in_toto` IMHO.

Corollary is that `_download_without_tuf` disappears, and `_download_with_tuf` gets privatised back to `__download_without_tuf`.

Not going to press too hard on this, but I think it reads cleaner and more accurately as a whole. Happy to make these changes, if you like.

Reply:
Done in e67611b.

I kept the call assertions for `_download_with_tuf` because it is sometimes called by other functions than `_download_without_tuf_in_toto` (for instance, by `__get_versions`)